### ISSUE TITLE
Error on any 'jspm' command: TypeError: Cannot read property 'startsWith' of undefined

### ISSUE THREAD

**Issue by hancsu** [jspm/jspm-cli#2148]
After successfully installing JSPM through NPM;

`npm install -g jspm`

any attempt to run 'jspm' on the command line or 'jspm SOMECOMMAND' all lead to the same error:

```
C:\some\path> jspm

C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\common.js:108
    if (pjson.startsWith('\uFEFF'))

TypeError: Cannot read property 'startsWith' of undefined
    at exports.readJSONSync (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\common.js:108:12)
    at Object.<anonymous> (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\global-config.js:53:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\registry.js:19:20)
```

or (another example):

```
C:\some\path> jspm registry config github

C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\common.js:108
    if (pjson.startsWith('\uFEFF'))

TypeError: Cannot read property 'startsWith' of undefined
    at exports.readJSONSync (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\common.js:108:12)
    at Object.<anonymous> (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\global-config.js:53:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\registry.js:19:20)
```

It's a fresh install of Node. Versions:

```
    C:\some\path> node -v
    v6.9.1
```

```
    C:\some\path> npm -v
    3.10.9
```

Link to StackOverflow thread: http://stackoverflow.com/questions/40358356/jspm-error-on-any-jspm-command-typeerror-cannot-read-property-startswith-o

EDIT: Out of interest I just tested out v0.10.0 and the 'jspm' command doesn't raise any error when run.

**Comment by hancsu** [comment]
Out of interest I just tested out v0.10.0 and the 'jspm' command doesn't raise any error when run.

**Comment by hancsu** [comment]
In "..\AppData\Roaming\npm\node_modules\jspm\lib\common.js", if I change Line 100:

`var pjson;` to `var pjson = null;`

It changes the error to:

```
C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\common.js:108
    if (pjson.startsWith('\uFEFF'))

TypeError: Cannot read property 'startsWith' of _**null**_
    at exports.readJSONSync (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\common.js:108:12)
    at Object.<anonymous> (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\global-config.js:53:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\registry.js:19:20)
```

**Comment by hancsu** [comment]
I added an if block round the offending line(s) to check whether pjson was defined:

```
if(pjson) {
    if (pjson.startsWith('\uFEFF')) {
        pjson = pjson.substr(1);
    }
}
```

And it gives the error:
```
C:\inetpub\wwwroot\minimal>jspm

C:\Users\8058\AppData\Roaming\npm\node_modules\jspm\lib\common.js:117
    throw 'Error parsing package.json file ' + file;
    ^
Error parsing package.json file \\.jspm\config
```

Which leads me to global-config.js where the `file` in:

`pjson = fs.readFileSync(file).toString();`

is defined. When I change the 'globalConfigFile' variable to the absolute path of 'package.json' from:

`var globalConfigFile = HOME + path.sep + '.jspm' + path.sep + 'config';`

to 

`var globalConfigFile = 'C:\\Users\\8058\\AppData\\Roaming\\npm\\node_modules\\jspm\\package.json';`

Running 'jspm' from the command line now works.

Not sure if that helps any.

**Comment by aluanhaddad** [comment]
There were some issues around windows paths a few versions back, I didn't encounter them myself, but you may have a corrupted installation. Have you tried reinstalling?

**Comment by hancsu** [comment]
Yeah I've re-installed Node and JSPM a couple of times with different versions and still no change.

**Comment by guybedford** [comment]
@hancsu you may find this is a permissions error that isn't throwing properly. I've added a fix in https://github.com/jspm/jspm-cli/commit/321df383e750ab031dc2a1cf99fc4262193e8c49 to properly throw the error. If you add that you should get a better message at least.

**Comment by hancsu** [comment]
I've added that fix to my global install and now the error reads as:

```
C:\>jspm
C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\lib\common.js:108
      throw e;
      ^

Error: UNKNOWN: unknown error, open '\\.jspm\config'
    at Error (native)
    at Object.fs.openSync (fs.js:640:18)
    at Object.fs.readFileSync (fs.js:508:33)
    at exports.readJSONSync (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm
\lib\common.js:102:16)
    at Object.<anonymous> (C:\Users\USERNAME\AppData\Roaming\npm\node_modules\jspm\l
ib\global-config.js:53:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
```

**Comment by guybedford** [comment]
@hancsu it seems like the `\\` is incorrect here. Note that that value is just `process.env.LOCALAPPDATA || process.env.HOME || process.env.HOMEPATH`. So you should be able to fix this by checking your environment variables I think?

**Comment by hancsu** [comment]
@guybedford I have `%APPDATA%` set to `C:\Users\USERNAME\AppData\Roaming` and `%LOCALAPPDATA%` set to `C:\Users\USERNAME\AppData\Local`.

`.jspm\config` exists in `C:\Users\USERNAME\`. Should it be somewhere else?

**Comment by guybedford** [comment]
@hancsu what I'm saying is that it seems like if you run `node -e "console.log(process.env.LOCALAPPDATA || process.env.HOME || process.env.HOMEPATH)"` on your machine that you would get `\` instead of `C:\Users\USERNAME\AppData\Local`? Perhaps you can check that?

**Comment by hancsu** [comment]
@guybedford Yeah I got '\'.

**Comment by guybedford** [comment]
Ensuring the environment variable is a valid absolute path should fix this then. The update throwing the correct error has been released in 0.16.48.
